### PR TITLE
test: tab bar functional tests

### PR DESCRIPTION
**Pull request by carlossteinzup**
### Related Issues
https://github.com/ZupIT/beagle/issues/500
https://github.com/ZupIT/beagle/issues/1014

### Description and Example

Adds functional/instrumented tests to the TabBar component

### Checklist

Please, check if these important points are met using `[x]`:

- [x] I read the [PR Guide] and followed the process outlined there for submitting this PR.
- [x] I avoided _breaking changes_ by not changing public APIs that people rely on. 
- [x] I am willing to follow-up on review comments in a timely manner.

[PR Guide]: https://github.com/ZupIT/beagle/blob/master/doc/contributing/pull_requests.md

### CODE REVIEW

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: iOS/Tests/AutomatedTestsUITests/Steps/SetContextSteps.swift [hunk 1]
```diff
@@ -55,25 +55,10 @@ class SetContextSteps: CucumberStepsDefinition {
     }
 }
 
-fileprivate enum Seeds: String {
+fileprivate enum SetContextScreenElements: String {
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/SetContextSteps.swift:58]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/ef60ccc42e1a85620ef91301f7ad0df265b9060c/iOS/Tests/AutomatedTestsUITests/Steps/SetContextSteps.swift#L58"></span>Prefer <code>private</code> over <code>fileprivate</code> declarations.<br />
<code>private_over_fileprivate</code> <code>SetContextSteps.swift:58</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/TabBarSteps.swift [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+import Foundation
+import XCTest
+
+class TabBarSteps: CucumberStepsDefinition {
+    
+    var application: XCUIApplication!
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/TabBarSteps.swift:22]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/ef60ccc42e1a85620ef91301f7ad0df265b9060c/iOS/Tests/AutomatedTestsUITests/Steps/TabBarSteps.swift#L22"></span>Implicitly unwrapped optionals should be avoided when possible.<br />
<code>implicitly_unwrapped_optional</code> <code>TabBarSteps.swift:22</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/TabBarSteps.swift [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        
+        Given("^that I'm on the tabBar screen$") { _, _ -> Void in
+            XCTAssertTrue(TabBarScreenElements.SCREEN_HEADER.element.exists)
+        }
+
+        // MARK: - When
+
+        // Scenario 2, 6
+        When(#"^I click on button "([^\"]*)" to change tab$"#) { args, _ -> Void in
+            let text = args![0]
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/TabBarSteps.swift:41]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/ef60ccc42e1a85620ef91301f7ad0df265b9060c/iOS/Tests/AutomatedTestsUITests/Steps/TabBarSteps.swift#L41"></span>Force unwrapping should be avoided.<br />
<code>force_unwrapping</code> <code>TabBarSteps.swift:41</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/TabBarSteps.swift [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+            self.application.buttons[text].tap()
+        }
+
+        // Scenario 3
+        When(#"^I click in a tab with text "([^\"]*)"$"#) { args, _ -> Void in
+            let text = args![0]
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/TabBarSteps.swift:47]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/ef60ccc42e1a85620ef91301f7ad0df265b9060c/iOS/Tests/AutomatedTestsUITests/Steps/TabBarSteps.swift#L47"></span>Force unwrapping should be avoided.<br />
<code>force_unwrapping</code> <code>TabBarSteps.swift:47</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/TabBarSteps.swift [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+            self.application.staticTexts[text].tap()
+        }
+        
+        // MARK: - Then
+        
+        // Scenario 1
+        Then(#"^I click on each tabBarItem and confirm its position$"#) { _, _ -> Void in
+            for i in 0...9 {
+                let tabTitle = "Tab\(i+1)"
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/TabBarSteps.swift:56]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/ef60ccc42e1a85620ef91301f7ad0df265b9060c/iOS/Tests/AutomatedTestsUITests/Steps/TabBarSteps.swift#L56"></span>Operators should be surrounded by a single whitespace when they are being used.<br />
<code>operator_usage_whitespace</code> <code>TabBarSteps.swift:56</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/TabBarSteps.swift [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+                let positionText = "Tab position \(i)"
+                self.application.staticTexts[tabTitle].tap()
+                XCTAssertTrue(self.application.staticTexts[positionText].exists)
+            }
+        }
+
+        // Scenario 2
+        Then(#"^the tab with text "([^\"]*)" must be on screen$"#) { args, _ -> Void in
+            let text = args![0]
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/TabBarSteps.swift:65]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/ef60ccc42e1a85620ef91301f7ad0df265b9060c/iOS/Tests/AutomatedTestsUITests/Steps/TabBarSteps.swift#L65"></span>Force unwrapping should be avoided.<br />
<code>force_unwrapping</code> <code>TabBarSteps.swift:65</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/TabBarSteps.swift [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+            XCTAssertTrue(self.application.staticTexts[text].exists)
+        }
+        
+        // Scenario 3
+        Then(#"^the tab position should have its text changed to "([^\"]*)"$"#) { args, _ -> Void in
+            let text = args![0]
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/TabBarSteps.swift:71]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/ef60ccc42e1a85620ef91301f7ad0df265b9060c/iOS/Tests/AutomatedTestsUITests/Steps/TabBarSteps.swift#L71"></span>Force unwrapping should be avoided.<br />
<code>force_unwrapping</code> <code>TabBarSteps.swift:71</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/TabBarSteps.swift [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+            XCTAssertTrue(self.application.staticTexts[text].exists)
+        }
+        
+        // Scenario 4, 6
+        Then(#"^check tab with "([^\"]*)" icon is on screen$"#) { args, _ -> Void in
+            let icon = args![0]
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/TabBarSteps.swift:77]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/ef60ccc42e1a85620ef91301f7ad0df265b9060c/iOS/Tests/AutomatedTestsUITests/Steps/TabBarSteps.swift#L77"></span>Force unwrapping should be avoided.<br />
<code>force_unwrapping</code> <code>TabBarSteps.swift:77</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/TabBarSteps.swift [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+            XCTAssertTrue(self.application.images[icon].exists)
+        }
+
+        // Scenario 5
+        Then(#"^check tab with text "([^\"]*)" and "([^\"]*)" icon are on screen$"#) { args, _ -> Void in
+            let text = args![0]
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/TabBarSteps.swift:83]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/ef60ccc42e1a85620ef91301f7ad0df265b9060c/iOS/Tests/AutomatedTestsUITests/Steps/TabBarSteps.swift#L83"></span>Force unwrapping should be avoided.<br />
<code>force_unwrapping</code> <code>TabBarSteps.swift:83</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/TabBarSteps.swift [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+            let icon = args![1]
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/TabBarSteps.swift:84]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/ef60ccc42e1a85620ef91301f7ad0df265b9060c/iOS/Tests/AutomatedTestsUITests/Steps/TabBarSteps.swift#L84"></span>Force unwrapping should be avoided.<br />
<code>force_unwrapping</code> <code>TabBarSteps.swift:84</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/TabBarSteps.swift [hunk 11]
```diff
@@ -0,0 +1,106 @@
...
+            XCTAssertTrue(self.application.staticTexts[text].exists)
+            XCTAssertTrue(self.application.images[icon].exists)
+        }
+        
+        // Scenario 6
+        Then(#"^the tab with text "([^\"]*)" and "([^\"]*)" icon will change icon to "([^\"]*)" icon$"#) { args, _ -> Void in
+            let text = args![0]
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/TabBarSteps.swift:91]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/ef60ccc42e1a85620ef91301f7ad0df265b9060c/iOS/Tests/AutomatedTestsUITests/Steps/TabBarSteps.swift#L91"></span>Force unwrapping should be avoided.<br />
<code>force_unwrapping</code> <code>TabBarSteps.swift:91</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/TabBarSteps.swift [hunk 12]
```diff
@@ -0,0 +1,106 @@
...
+            let changedIcon = args![2]
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/TabBarSteps.swift:92]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/ef60ccc42e1a85620ef91301f7ad0df265b9060c/iOS/Tests/AutomatedTestsUITests/Steps/TabBarSteps.swift#L92"></span>Force unwrapping should be avoided.<br />
<code>force_unwrapping</code> <code>TabBarSteps.swift:92</code></td>
    </tr>
  </tbody>
</table>

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          5 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/ef60ccc42e1a85620ef91301f7ad0df265b9060c/iOS/Tests/AutomatedTestsUITests/Steps/SetContextSteps.swift#L18">iOS/Tests/AutomatedTestsUITests/Steps/SetContextSteps.swift#L18</a> - Limit vertical whitespace to a single empty line. Currently 2.<br />
<code>vertical_whitespace</code> <code>SetContextSteps.swift:18</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/ef60ccc42e1a85620ef91301f7ad0df265b9060c/iOS/Tests/AutomatedTestsUITests/Steps/SetContextSteps.swift#L23">iOS/Tests/AutomatedTestsUITests/Steps/SetContextSteps.swift#L23</a> - Implicitly unwrapped optionals should be avoided when possible.<br />
<code>implicitly_unwrapped_optional</code> <code>SetContextSteps.swift:23</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/ef60ccc42e1a85620ef91301f7ad0df265b9060c/iOS/Tests/AutomatedTestsUITests/Steps/SetContextSteps.swift#L42">iOS/Tests/AutomatedTestsUITests/Steps/SetContextSteps.swift#L42</a> - Opening braces should be preceded by a single space and on the same line as the declaration.<br />
<code>opening_brace</code> <code>SetContextSteps.swift:42</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/ef60ccc42e1a85620ef91301f7ad0df265b9060c/iOS/Tests/AutomatedTestsUITests/Steps/SetContextSteps.swift#L43">iOS/Tests/AutomatedTestsUITests/Steps/SetContextSteps.swift#L43</a> - Force unwrapping should be avoided.<br />
<code>force_unwrapping</code> <code>SetContextSteps.swift:43</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/ef60ccc42e1a85620ef91301f7ad0df265b9060c/iOS/Tests/AutomatedTestsUITests/Steps/SetContextSteps.swift#L51">iOS/Tests/AutomatedTestsUITests/Steps/SetContextSteps.swift#L51</a> - Force unwrapping should be avoided.<br />
<code>force_unwrapping</code> <code>SetContextSteps.swift:51</code></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_ios">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/TabBarScreenBuilder.kt [hunk 13]
```diff
@@ -16,79 +16,85 @@
...
+import br.com.zup.beagle.widget.ui.Button
 import br.com.zup.beagle.widget.ui.ImagePath
 import br.com.zup.beagle.widget.ui.TabBar
 import br.com.zup.beagle.widget.ui.TabBarItem
 import br.com.zup.beagle.widget.ui.Text
 
+data class TabImage(val mobileImageId: String, val webImageUrl: String)
+
+const val WEB_IMAGE_PATH = "https://www.petlove.com.br/images/breeds/193436/profile/original/beagle-p.jpg"
+const val WEB_IMAGE_PATH2 = "https://i.picsum.photos/id/237/200/300.jpg?hmac=TmmQSbShHz9CdQm0NkEjx1Dyh_Y984R9LpNrpvH2D_U"
```
Comment (botbeaglezup) [inline backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/TabBarScreenBuilder.kt:33]: <table data-meta="generated_by_backend">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/e40206528f9641bf75cbd1afdd0c70175694b3a2/backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/TabBarScreenBuilder.kt#L33"></span>Line detected that is longer than the defined maximum line length in the code style.</td>
    </tr>
  </tbody>
</table>

**Review by lucasfonsecazup — APPROVED** [review]

File: backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/TabBarScreenBuilder.kt [hunk 14]
```diff
@@ -16,79 +16,85 @@
...
+import br.com.zup.beagle.widget.ui.Button
 import br.com.zup.beagle.widget.ui.ImagePath
 import br.com.zup.beagle.widget.ui.TabBar
 import br.com.zup.beagle.widget.ui.TabBarItem
 import br.com.zup.beagle.widget.ui.Text
 
+data class TabImage(val mobileImageId: String, val webImageUrl: String)
+
+const val IMAGE_PATH = "https://www.petlove.com.br/images/breeds/193436/profile/original/beagle-p.jpg"
+const val IMAGE_PATH2 = "https://i.picsum.photos/id/237/200/300.jpg?hmac=TmmQSbShHz9CdQm0NkEjx1Dyh_Y984R9LpNrpvH2D_U"
```
Comment (paulomeurerzup) [inline backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/TabBarScreenBuilder.kt:33]: Is it a good idea to point to urls that could be broken in future? I think that creating an endpoint on tests backend to serve this images is a more reliable solution.

File: android/automated-tests/src/androidTest/kotlin/br/com/zup/beagle/automatedTests/utils/matcher/MatcherExtension.kt [hunk 15]
```diff
@@ -15,17 +15,16 @@
...
 import android.view.View
 import android.widget.TextView
-import androidx.test.espresso.AmbiguousViewMatcherException
-import androidx.test.espresso.NoMatchingRootException
-import androidx.test.espresso.NoMatchingViewException
-import androidx.test.espresso.UiController
-import androidx.test.espresso.ViewAction
-import androidx.test.espresso.ViewInteraction
+import androidx.core.content.ContextCompat
+import androidx.test.espresso.*
```
Comment (hthemir) [inline android/automated-tests/src/androidTest/kotlin/br/com/zup/beagle/automatedTests/utils/matcher/MatcherExtension.kt:25]: Remove wildcard import

File: backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/controllers/GeneralController.kt [hunk 16]
```diff
@@ -30,8 +30,17 @@ import br.com.zup.beagle.automatedtests.builders.SafeAreaScreenBuilder
 import br.com.zup.beagle.automatedtests.builders.BuilderKotlinDslScreenBuilder
 import br.com.zup.beagle.automatedtests.builders.ExpressionEscapingScreenBuilder
 import br.com.zup.beagle.automatedtests.builders.AccessibilityScreenBuilder
+import br.com.zup.beagle.automatedtests.constants.IMAGE_WEB
+import org.springframework.core.io.ClassPathResource
+import org.springframework.core.io.InputStreamResource
+import org.springframework.http.MediaType
+import org.springframework.http.ResponseEntity
+import org.springframework.util.StreamUtils
```
Comment (botbeaglezup) [inline backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/controllers/GeneralController.kt:38]: <table data-meta="generated_by_backend">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/8ac5cbfa6bcb73fe8d5986e99d9083732e16eace/backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/controllers/GeneralController.kt#L38"></span>The import ‘org.springframework.util.StreamUtils’ is unused.</td>
    </tr>
  </tbody>
</table>

File: backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/controllers/GeneralController.kt [hunk 17]
```diff
@@ -30,8 +30,17 @@ import br.com.zup.beagle.automatedtests.builders.SafeAreaScreenBuilder
...
 import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.RequestParam
```
Comment (botbeaglezup) [inline backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/controllers/GeneralController.kt:41]: <table data-meta="generated_by_backend">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/8ac5cbfa6bcb73fe8d5986e99d9083732e16eace/backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/controllers/GeneralController.kt#L41"></span>The import ‘org.springframework.web.bind.annotation.RequestParam’ is unused.</td>
    </tr>
  </tbody>
</table>

File: backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/controllers/GeneralController.kt [hunk 18]
```diff
@@ -30,8 +30,17 @@ import br.com.zup.beagle.automatedtests.builders.SafeAreaScreenBuilder
...
 import org.springframework.web.bind.annotation.RestController
+import javax.servlet.http.HttpServletResponse
```
Comment (botbeaglezup) [inline backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/controllers/GeneralController.kt:43]: <table data-meta="generated_by_backend">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/8ac5cbfa6bcb73fe8d5986e99d9083732e16eace/backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/controllers/GeneralController.kt#L43"></span>The import ‘javax.servlet.http.HttpServletResponse’ is unused.</td>
    </tr>
  </tbody>
</table>

File: backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/controllers/GeneralController.kt [hunk 19]
```diff
@@ -30,7 +30,13 @@ import br.com.zup.beagle.automatedtests.builders.SafeAreaScreenBuilder
 import br.com.zup.beagle.automatedtests.builders.BuilderKotlinDslScreenBuilder
 import br.com.zup.beagle.automatedtests.builders.ExpressionEscapingScreenBuilder
 import br.com.zup.beagle.automatedtests.builders.AccessibilityScreenBuilder
+import br.com.zup.beagle.automatedtests.constants.IMAGE_WEB
+import org.springframework.core.io.ClassPathResource
+import org.springframework.core.io.InputStreamResource
+import org.springframework.http.MediaType
+import org.springframework.http.ResponseEntity
 import org.springframework.web.bind.annotation.GetMapping
```
Comment (botbeaglezup) [inline backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/controllers/GeneralController.kt:38]: <table data-meta="generated_by_backend">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/6e7eee89803750f280d816678eb16a087d6f3c71/backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/controllers/GeneralController.kt#L38"></span>The import ‘org.springframework.util.StreamUtils’ is unused.</td>
    </tr>
  </tbody>
</table>

File: backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/controllers/GeneralController.kt [hunk 20]
```diff
@@ -30,7 +30,13 @@ import br.com.zup.beagle.automatedtests.builders.SafeAreaScreenBuilder
...
+import org.springframework.web.bind.annotation.PathVariable
 import org.springframework.web.bind.annotation.RestController
```
Comment (botbeaglezup) [inline backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/controllers/GeneralController.kt:41]: <table data-meta="generated_by_backend">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/6e7eee89803750f280d816678eb16a087d6f3c71/backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/controllers/GeneralController.kt#L41"></span>The import ‘org.springframework.web.bind.annotation.RequestParam’ is unused.</td>
    </tr>
  </tbody>
</table>

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/6e7eee89803750f280d816678eb16a087d6f3c71/backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/controllers/GeneralController.kt#L43">backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/controllers/GeneralController.kt#L43</a> - The import ‘javax.servlet.http.HttpServletResponse’ is unused.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_backend">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/TabBarScreenBuilder.kt (hunk 14 above)
Comment (carlossteinzup) [reply backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/TabBarScreenBuilder.kt]: Yes, ideed. We have changed the path to a local and have set the image resources on the BFF.

**Review by lucasfonsecazup — APPROVED** [review]

**Review by paulomeurerzup — APPROVED** [review]

File: backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/controllers/GeneralController.kt (hunk 19 above)
Comment (carlossteinzup) [reply backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/controllers/GeneralController.kt:38]: removed

**Review by uziasferreirazup — APPROVED** [review]

**Pull request merged**